Glossary (#7)
tomeshnet/toronto-community-network-docs: src/glossary.md

# Glossary

> ## Toronto Community Network

Reviewer: Should we also include acronyms like "tcn"

Reviewer 2: That acronym is not formalized, so I left it for now.

Reviewer: Perhaps it needs to be,

It is being used in FQDN naming
https://github.com/tomeshnet/toronto-community-network/issues/21

> For example a FQDN (fully qualified domain name) will be sn1a1.core.tcn.tomesh.net for a device operated by the core team at Toronto Community Network. The node will also answer as sn1a1.tcn.tomesh.net

Reviewer 2: Yes. I was sort of waiting for someone else to formalize it, but I'll add it here.

The name of the Toronto-based community mesh network initiative started by Toronto Mesh with its collaborators in 2020.

This book documents the current resources of this initiative. For more information, visit [the project page](https://github.com/tomeshnet/toronto-community-network/).

## Toronto Mesh

A community group that helps communities create better networks with open source and peer-to-peer technologies that promote digital literacy and privacy.

The group that started the Toronto Community Network initiative and its steward for the foreseeable future. For more information, visit [tomesh.net](https://tomesh.net).

## Point-to-point (PTP)

> A **PTP** link is a connection that only involves two antennas. These kinds of antennas are more powerful and can reach longer distances, but only one other antenna can connect to them. They're usually implemented using highly directional (i.e. high gain) antennas and used to connect two supernodes.

Reviewer: This is assuming wireless.  PTP also refers to entities likes Tunnels and vpns
Example: tun0: flags=4305<UP,**_POINTOPOINT_**,RUNNING,NOARP,MULTICAST>  mtu 1500

Also PTP does not define the power of the antenna.  The current antennas we have can be both PTP and PTMP.  The only differnce between the two is the speed as PTP can do 80Mhz while PTMP can only do 40.

The power of the antennas is defined by the geometry of the beam not  PTP vs  PTMP

IE: the LAP 120 with a 120 degree beam can act as PTP mode or PTMP mode, so can the LiteBeam whith only a 24 degree beam.

Reviewer: ```suggestion
A **PTP** (Point To Point) link is a connection limited to only two devices.  This could be a virtual connection like a VPN or Tunnel, or a physical connection like two routers connected via a fibre optic link, or two antennas configured for P2P.  Although most antennas are capable of doing both PTP and PTMP,  antennas used for PTP usually are high gain by being directional.  This allows them to reach longer distances with less noise.

## Point-to-multipoint (PTMP)

> **PTMP** antennas support multiple links by allowing many antennas to connect to it. They're usually sector antennas (moderate gain) installed at a high point. These kinds of links are used to connect homes on the ground, because one antenna may connect to up to hundreds of others pointing to it. These kinds of links are more limited in range than PTP setups.

Reviewer: see https://github.com/tomeshnet/toronto-community-network-docs/pull/7#issuecomment-678828732

## Node

A **node** is a device that's connected and accessible on the mesh network. This might be a router and antenna combo, for example, on top of someone's house, or an apartment building. This doesn't include devices connected to the mesh network that are [NATed](https://en.wikipedia.org/wiki/Network_address_translation), so even if your home router is connected to the mesh network, your laptop and phone usually aren't counted as nodes.

## Supernode

A **supernode** is an actively maintained relaying node that supports high bandwidth point-to-multipoint (PTMP) connectivity. Currently all supernodes are managed by Toronto Mesh, but other groups may also operate their own supernodes on the Toronto Community Network.

## Babel

> The **Babel routing protocol** is what we use to create the mesh network. It's the software that allows the "meshing" to happen, by routing packets where they need to go.

Reviewer: Babel does not actaully route any packets. The router does that. Babel dynamically installs routes into the operating system after deciding which direction is the best next hop  packets should take for each destination/subnet.

Once the routes are installed, the operating system uses them as if they where provided by any other source.

Reviewer 2: Yes, this is a simplification, as I figured many people will not understand routing rules or route tables. If you have a suggested change I'd be happy to look at it but I'm okay with this simplification.

Reviewer: My issue is that this simplification presents babel as something it is not - a router. IE CJDNS is a router. Babel has no interaction with the packets that cross the mesh. In fact the underlying can told not to list for babel.

The line also confuses BABEL (a protocol) with BABELD (a software). (HTTP is a protocol. Firefox is software)

If we are looking for simplifications it would be 
`The **Babel routing protocol** is a protocol for create a layer 3 mesh network. `

I don't know how to summarize it in a line or two. I think i need to expand it on my babeld page i been writing.

But maybe.

```suggestion
The **Babel routing protocol** is a protocol that is used to create a layer 3 mesh network. By running a babel implementations on nodes of a network, routes (the next direction a packet should take) to all the nodes will be actively calculated and implement on those nodes.
```

Reviewer 2: I committed this one, what do you think?
> `The **Babel routing protocol** is used to create a mesh network. By running a Babel implementations on the nodes of the network, paths to all other nodes are actively calculated, so that connections can be made to any node on the mesh.`

[Wikipedia](https://en.wikipedia.org/wiki/Babel_(protocol)), [Homepage](https://www.irif.fr/~jch/software/babel/)

> ## babled

Reviewer: ```suggestion
## babeld
```

> Pronounced "bable-dee", this is a reference implementation of the Bable protocol, and the software used to run Babel on the mesh network. The "d" stands for [daemon](https://en.wikipedia.org/wiki/Daemon_%28computing%29). The code is available [here](https://github.com/jech/babeld).

Reviewer: ```suggestion
Pronounced "babel-dee", this is a reference implementation of the Babel protocol, and the software used to run Babel on the mesh network. The "d" stands for [daemon](https://en.wikipedia.org/wiki/Daemon_%28computing%29). The code is available [here](https://github.com/jech/babeld).
```
